### PR TITLE
Add integration test for GRM secret controller

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area testing
/kind enhancement

**What this PR does / why we need it**:
Add integration test for resource-manager secret controller. Checks basic functionality i.e. finalizers on the secret are added/removed correctly.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/4805

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: test/integration/resourcemanager/secret/secret_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		AfterEach(func() {
+			Expect(testClient.Delete(ctx, managedResource)).To(Or(Succeed(), BeNotFoundError()))
+			Expect(testClient.Delete(ctx, secret)).To(Or(Succeed(), BeNotFoundError()))
+			// Wait for clean up of the secret
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)
+			}).Should(BeNotFoundError())
+		})
+
+		It("should successfully add finalizer to all the secrets reference by managed resource", func() {
```
Comment (shafeeqes) [inline test/integration/resourcemanager/secret/secret_test.go]:
```suggestion
		It("should successfully add finalizer to all the secrets which are referenced by ManagedResources", func() {
```

File: test/integration/resourcemanager/secret/secret_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+			Eventually(func(g Gomega) []string {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)).To(Succeed())
+				return secret.ObjectMeta.Finalizers
+			}).Should(
+				ContainElement(finalizerName),
+			)
+		})
+
+		It("should remove finalizer from secret which are no longer referenced by any managed resource", func() {
```
Comment (shafeeqes) [inline test/integration/resourcemanager/secret/secret_test.go]:
```suggestion
		It("should remove finalizer from secrets which are no longer referenced by any ManagedResource", func() {
```

File: test/integration/resourcemanager/secret/secret_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+			Eventually(func(g Gomega) []string {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)).To(Succeed())
+				return secret.ObjectMeta.Finalizers
+			}).ShouldNot(
+				ContainElement(finalizerName),
+			)
+		})
+
+		It("should do nothing if there is no MR refrencing the secret", func() {
```
Comment (shafeeqes) [inline test/integration/resourcemanager/secret/secret_test.go]:
```suggestion
		It("should do nothing if there is no ManagedResource referencing the secret", func() {
```

**Review by shafeeqes — COMMENTED** [review]
Minor nits, otherwise lgtm.

File: test/integration/resourcemanager/secret/secret_test.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		AfterEach(func() {
+			Expect(testClient.Delete(ctx, managedResource)).To(Or(Succeed(), BeNotFoundError()))
+			Expect(testClient.Delete(ctx, secret)).To(Or(Succeed(), BeNotFoundError()))
+			// Wait for clean up of the secret
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)
+			}).Should(BeNotFoundError())
+		})
+
+		It("should successfully add finalizer to all the secrets reference which are referenced by ManagedResources", func() {
```
Comment (shafeeqes) [inline test/integration/resourcemanager/secret/secret_test.go]:
```suggestion
		It("should successfully add finalizer to all the secrets which are referenced by ManagedResources", func() {
```

File: test/integration/resourcemanager/secret/secret_test.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		It("should do nothing if there is no ManagedResource referencing the secret", func() {
+			secret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "bar",
+					Namespace: testNamespace.Name,
+				},
+			}
+			Expect(testClient.Create(ctx, secret)).To(Succeed())
+
+			Eventually(func(g Gomega) []string {
```
Comment (rfranzke) [inline test/integration/resourcemanager/secret/secret_test.go]:
```suggestion
			Consistently(func(g Gomega) []string {
```

File: test/integration/resourcemanager/secret/secret_test.go [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+	BeforeEach(func() {
+		managedResource = &resourcesv1alpha1.ManagedResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    testNamespace.Name,
+				GenerateName: "test-",
+			},
+			Spec: resourcesv1alpha1.ManagedResourceSpec{
+				SecretRefs: []corev1.LocalObjectReference{{
+					Name: "foo",
+				}},
```
Comment (rfranzke) [inline test/integration/resourcemanager/secret/secret_test.go]: Maybe add two secrets here to improve the test accuracy?

File: test/integration/resourcemanager/secret/secret_test.go [hunk 7]
```diff
@@ -32,7 +32,8 @@ const (
 var _ = Describe("Secret controller tests", func() {
 	var (
 		managedResource *resourcesv1alpha1.ManagedResource
-		secret          *corev1.Secret
+		secret_foo      *corev1.Secret
+		secret_bar      *corev1.Secret
```
Comment (rfranzke) [inline test/integration/resourcemanager/secret/secret_test.go]: Nit: use CamelCase

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**